tests/transforms/preprocessing/test_pad.py

```diff
@@ -24,3 +24,13 @@ def test_nans_history(self):
         padded = tio.Pad(1, padding_mode=2)(self.sample_subject)
         again = padded.history[0](self.sample_subject)
         assert not torch.isnan(again.t1.data).any()
+
+    def test_padding_modes(self):
+        def padding_func():
```

Comment:
How is padding determined in this case? Could we have an example with a callable that actually does something?

Reply:
What do you mean by "How is padding determined in this case"? Shouldn't it just be padding =10? Or maybe the example makes it more understandable:
```        
def padding_func(vector, iaxis_pad_width, iaxis, kwargs):
       vector[:iaxis_pad_width[0]] = np.random.randint(0, 100, size=iaxis_pad_width[0])
       if iaxis_pad_width[1]>0:
          vector[-iaxis_pad_width[1]:] = np.random.randint(0, 100, size=iaxis_pad_width[1])
```

Reply:
I just mean how are the values in the padded voxels computed when the padding function does nothing. I tried your snippet above, but it seemed to hang (or just be very slow):

```python
import numpy as np
import torchio as tio

def padding_func(vector, iaxis_pad_width, iaxis, kwargs):
    vector[:iaxis_pad_width[0]] = np.random.randint(0, 100, size=iaxis_pad_width[0])
    if iaxis_pad_width[1]>0:
        vector[-iaxis_pad_width[1]:] = np.random.randint(0, 100, size=iaxis_pad_width[1])

pad = tio.Pad(50, padding_func)
t1 = tio.datasets.FPG().t1
p = pad(t1)
```

Reply:
Yes, it is super slow for a "real" example and only works for small arrays with few entries. However, I think it's the numpy "pad" function and not the code. Numpy creates padding with zeros by default (https://github.com/numpy/numpy/blob/54c52f13713f3d21795926ca4dbb27e16fada171/numpy/lib/arraypad.py#L745). If the function does nothing, as in the example, this corresponds to zero padding.